File: backend/src/main/java/it/unical/inf/ea/backend/dto/DeliveryDTO.java
package it.unical.inf.ea.backend.dto;

import it.unical.inf.ea.backend.dto.enums.DeliveryStatus;
import jakarta.validation.constraints.NotNull;
import lombok.*;
import org.hibernate.validator.constraints.Length;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class DeliveryDTO {
    @NotNull
    private String Id;

    @NotNull
    private LocalDateTime sendTime;


    private LocalDateTime deliveredTime;


    private BigDecimal deliveryCost;

    @Length(max = 50)
    private String shipper;

    @NotNull
    private DeliveryStatus deliveryStatus;

    @NotNull
    private AddressDTO senderAddress;

    @NotNull
    private AddressDTO receiverAddress;
}

File: frontend/app/src/main/java/com/android/frontend/dto/DeliveryDTO.kt
package com.android.frontend.dto

import java.math.BigDecimal
import java.time.LocalDateTime

data class DeliveryDTO(
    var id: String,
    var sendTime: LocalDateTime,
    var deliveredTime: LocalDateTime? = null,
    var deliveryCost: BigDecimal? = null,
    var shipper: String? = null,
    var deliveryStatus: DeliveryStatus,
    var senderAddress: AddressDTO,
    var receiverAddress: AddressDTO
){
    enum class DeliveryStatus {
        SHIPPED,
        DELIVERED
    }
}